### ISSUE TITLE
Support Mac installation

### ISSUE THREAD

**Issue by zbloss** [triton-inference-server/pytriton#44]
It would be great to be able to install pytriton on Macs for ease-of-development. Even with the lack of CUDA support for Macs, being able to develop using only the CPU would be a real time saver.

As a far reaching stretch goal, being able to use `mps` on Macs would also be great.

As of now `pip install nvidia-pytriton` throws an error being unable to install `cuda-python` on my machine.

My Machine:
* Apple M2 Max
* 32GB Memory
* Year - 2023
* Python - 3.11.6

**Comment by sricke** [comment]
As far as I understand, that update should come from triton-inference-server, but since it's supposed to run on an NDVIDIA GPU I don't think it'll happen.

What you can do is run on a docker emulated image running docker image with flag `--platform linux/amd64`. It will run on CPU and slower but you'll be able to debug. 

Hope that helps

**Comment by zbloss** [comment]
Thanks for the quick fix, that's what I'll start with.

Is it possible to decouple pytriton from the cuda-python library?

**Comment by jkosek** [comment]
The `cuda-python` is required by `tritonclient` package which is PyTriton dependency. As for now we are not able to remove it, but if you would run installation inside the Docker container with Ubuntu 22.04 there should be no issue despite the CUDA is not available in the system.

**Comment by zbloss** [comment]
Thanks all!

**Comment by martin-liu** [comment]
@jkosek It seems `cuda-python` is [optional](https://github.com/triton-inference-server/client/blob/main/src/python/library/setup.py#L70-L76) for `tritonclient`, is it possible to make it optional in `pytriton`?

**Comment by jkosek** [comment]
@martin-liu you are correct. We are going to fix dependencies on our side to avoid cuda dependency at the moment. Thanks!

**Comment by martin-liu** [comment]
@jkosek, fantastic! I'm truly appreciative of your efforts to resolve this issue. This adjustment will significantly enhance local development experience on Mac. Thank you!

**Comment by jkosek** [comment]
@martin-liu the change has been applied on the main branch and will be available with next wheel release.

**Comment by martin-liu** [comment]
@jkosek Great! Thank you very much!

**Comment by martin-liu** [comment]
@jkosek may I know what's the ETA for next release?

**Comment by martin-liu** [comment]
@jkosek Thank you for the update, I'll check out!

**Comment by martin-liu** [comment]
@jkosek Seems there are no wheels available that support macOS. Are there any specific issues or blockers preventing the creation of macOS wheels?

**Comment by jkosek** [comment]
Hey @martin-liu . At the moment we are not able to provide a macOS wheel. The PyTriton contains a Triton binaries which are bounded to the Linux OS and specific version of GLibc. In such case the only option we can suggest is to use Docker image, ex. Ubuntu, on macOS.

**Comment by martin-liu** [comment]
@jkosek got it, thanks

**Comment by dineshtrivedi** [comment]
@jkosek @martin-liu , what do you think about creating a client lib spliting python code and server binary?

I tried avoiding dependencies and code duplication, happy to hear thoughts and suggestions.

https://github.com/triton-inference-server/pytriton/pull/117

**Comment by dineshtrivedi** [comment]
Hi @jkosek , just a ping on about the PR above. Could you please assist? Or should I reach out to someone else?

**Comment by jkosek** [comment]
Hi @dineshtrivedi  - apologies for the delayed response.

We don’t have plans to support macOS or Windows for PyTriton or Triton.

At this time we’re only accepting critical fixes for PyTriton, as the long‑term plan is to migrate its functionality into the Dynamo project: https://github.com/ai-dynamo.

**Comment by dineshtrivedi** [comment]
Thanks for the response, @jkosek . I didn't know of Dynamo. I will have a look.
